azurelinuxagent/ga/exthandlers.py

```diff
@@ -1350,13 +1350,25 @@ def get_handler_status(self):
         if not os.path.isfile(status_file):
             return None
 
+        handler_status_contents = ""
         try:
-            data = json.loads(fileutil.read_file(status_file))
+            handler_status_contents = fileutil.read_file(status_file)
+            data = json.loads(handler_status_contents)
             handler_status = ExtHandlerStatus()
             set_properties("ExtHandlerStatus", handler_status, data)
             return handler_status
         except (IOError, ValueError) as e:
             self.logger.error("Failed to get handler status: {0}", e)
+        except Exception as e:
+            error = "Failed to get handler status message: {0}. \n Contents of file: {1}".format(
+                ustr(e),
+                handler_status_contents)
```

Comment:
rather, we need 'data'

also, please double-check the event makes it to kusto since the telemetry pipeline seems to have issues with some json content

Reply:
`data` is the one failing (especially in the scenario that we want to debug. Fetching the HandlerStatus contents from file works fine but as soon as we try to convert it into JSON that's when it throws - 
```python
Exception processing extension handlers: too many digits in integer
<FrameSummary file bin/WALinuxAgent-2.2.46-py2.7.egg/azurelinuxagent/ga/exthandlers.py, line 266 in run>
<FrameSummary file bin/WALinuxAgent-2.2.46-py2.7.egg/azurelinuxagent/ga/exthandlers.py, line 583 in report_ext_handlers_status>
<FrameSummary file bin/WALinuxAgent-2.2.46-py2.7.egg/azurelinuxagent/ga/exthandlers.py, line 660 in report_ext_handler_status>
<FrameSummary file bin/WALinuxAgent-2.2.46-py2.7.egg/azurelinuxagent/ga/exthandlers.py, line 1363 in get_handler_status>
<FrameSummary file /usr/lib/python3.5/json/__init__.py, line 319 in loads>
<FrameSummary file /usr/lib/python3.5/json/decoder.py, line 339 in decode>
<FrameSummary file /usr/lib/python3.5/json/decoder.py, line 355 in raw_decode>]
```

Reply:
>also, please double-check the event makes it to kusto since the telemetry pipeline seems to have issues with some json content

Good point, I'll do a manual run to see if it makes through to Kusto

Reply:
What exception does it throw? Is it `ValueError` or `DecodeError`? Also what happens if the issue is persistent - i.e. if the status file isn't updated and stays in that form, would you be sending events every time you hit it? which could be every 3 seconds?

Also, It should be already taken care of by https://github.com/Azure/WALinuxAgent/pull/1761, with `_read_and_parse_json_status_file()`, called by `collect_ext_status()`, along with the Telemetry event sent if any error comes.

Reading and Parsing the json, with exception handling:
https://github.com/Azure/WALinuxAgent/blob/6242409be125ab539ef55393aa23527d70f32385/azurelinuxagent/ga/exthandlers.py#L1464-L1481

Which is caught, packaged and sent to telemetry: https://github.com/Azure/WALinuxAgent/blob/6242409be125ab539ef55393aa23527d70f32385/azurelinuxagent/ga/exthandlers.py#L1115-L1128

Reply:
>data is the one failing
Yup, misread the code, you are good

Reply:
@vrdmr - #1761 is for the extension's *.sattus file. This one is for the agent's HandlerStatus. Also, the way to handle the error is different.